doc/content/system/scheduler.md

```diff
@@ -1,4 +1,68 @@
+The animation below illustrates how the simple scheduler sends work to a device.
+![simple scheduler animation](animation.gif "Animation of the simple scheduler")
+
+This animation shows the dispatcher sending subbatches to the device until it reaches capacity.  The dispatcher then blocks until the device completes a subbatch, at which it sends another subbatch to fill the device back to capacity.  This pattern of work continues until the scheduler and dispatcher complete the entire, original batch of work.
+
+This scheduling algorithm is useful:
+1. To determine the optimal subbatch size for a single device.  Typically there is an optimal subbatch size for a given model.  The size will depend on the model, the delay in sending data to the device, and the device throughput.  In practice it's best to determine that optimal size empircally by a numerical experiment and then reuse the optimal batch size in the more sophisticated scheduling algorithms.
+2. In cases where each NEML2 instance, for example each MPI rank, only has exclusive access to a single accelerator.  For example, running one MPI rank per node and each node has a single GPU.
+
+It also demonstrates the basic concepts applied in other schedulers.
+
+### SimpleMPIScheduler
+
+The [SimpleMPIScheduler](#simplempischeduler) is an extension of the [SimpleScheduler](#simplescheduler) which better distributes work in MPI parallel applications calling NEML2.  This scheduler assigns each MPI rank a unique accelerator device out of a list of devices provided by the user.  This MPI rank sends work to its assigned device exclusively, using the same algorithm for work scheduling as in the [SimpleScheduler](#simplescheduler).  The difference in inputs between the [SimpleMPIScheduler](#simplempischeduler) and the [SimpleScheduler](#simplescheduler) is just that the user now provides a list of devices, a list of subbatch sizes, and (optionally), a list of capacities.  Each item in the list is a single device available to MPI ranks.
+
+This scheduler assigns a rank to a device by first forming a local MPI communicator over the MPI ranks running on a single compute node (determined by the hostname provided by the [hwloc library](https://www.open-mpi.org/projects/hwloc/)).  Each MPI rank in the communicator is assigned one device from the list provided by the user.  The user must provide at least as many devices as the local communicator has MPI ranks.  This approach limits the number of MPI ranks a user can launch *per node* to the number of available accelerators.  For example, if a compute node has one CPU with 16 threads and 8 GPU accelerators the host application can only run 8 MPI ranks per node.  The scheduler will report an error if the host application tries to run it with more local ranks than user supplied devices.  This often means the user will need to [pin ranks](https://hpc-wiki.info/hpc/Binding/Pinning) to ensure their MPI application assigns only the required number of ranks per node.
```

Comment:
Now that I think more about it, if we only use hwloc to get the hostname, why not just use `MPI_Get_processor_name` provided by the MPI implementation plus an all-to-all?